vms/platformvm/vm.go

```diff
@@ -393,7 +393,9 @@ func (vm *VM) LastAccepted(context.Context) (ids.ID, error) {
 
 // SetPreference sets the preferred block to be the one with ID [blkID]
 func (vm *VM) SetPreference(_ context.Context, blkID ids.ID) error {
-	vm.Builder.SetPreference(blkID)
+	if vm.manager.SetPreference(blkID) {
+		vm.Builder.ResetBlockTimer()
```

Comment:
nit: I wonder if we should rename this method.  We reset the timer here to check whether there are new blocks ready to be issued. Should we rename this to `TryCreateBlock` or something? Note that this is also called when mempool.Add(tx) completes

Reply:
Gonna retain it for now to keep the diff minimal- as we discussed on our call, we probably want to rethink how we're building blocks to prevent unnecessary work